spec/unit/cli/cli_spec.rb

```diff
@@ -0,0 +1,84 @@
+# encoding: utf-8
+
+require 'spec_helper'
+require 'test_utils'
+
+
+describe Hocon::CLI do
+  ####################
+  # Argument Parsing
+  ####################
+  context 'argument parsing' do
+    it 'should find all the flags and arguments' do
+      args = %w(-i foo -o bar set some.path some_value --json)
+      expected_options = {
+        in_file: 'foo',
+        out_file: 'bar',
+        subcommand: 'set',
+        path: 'some.path',
+        new_value: 'some_value',
+        json: true
+      }
+      expect(Hocon::CLI.parse_args(args)).to eq(expected_options)
+    end
+  end
+
+  context 'subcommands' do
+    hocon_text =
+'foo.bar {
+  baz = 42
+  array = [1, 2, 3]
+  hash: {key: value}
+}'
+
+    context 'do_get()' do
+      it 'should get simple values' do
+        options = {path: 'foo.bar.baz'}
+        expect(Hocon::CLI.do_get(options, hocon_text)).to eq('42')
+      end
+
+      it 'should work with arrays' do
+        options = {path: 'foo.bar.array'}
+        expected = "[\n    1,\n    2,\n    3\n]"
+        expect(Hocon::CLI.do_get(options, hocon_text)).to eq(expected)
+      end
+
+      it 'should work with hashes' do
+        options = {path: 'foo.bar.hash'}
+        expected = "key=value\n"
```

Comment:
Would probably be at least worth tracking down where the `=` comes from.  As for the outer braces, I don't know that I totally care whether or not we have them for now, except that if we don't have them now and we want to add them in the future it might be considered an API breakage.  On the fence as to whether or not the output makes more sense w/or w/o the braces.  :/  Do you have any thoughts?  I mean, I guess if someone was redirecting the output to a file, then without the braces is what we'd prefer?